auth/downscoped/credential_initialization.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package downscopingoverview contains Google Cloud auth snippets showing how to
+// downscope credentials with Credential Access Boundaries.
+// https://cloud.google.com/iam/docs/downscoping-short-lived-credentials
+package downscopedoverview
+
+// [START auth_downscoping_initialize_downscoped_cred]
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/google/downscope"
+)
+
+// initializeCredentials will generate a downscoped token using the provided Access Boundary Rules.
+//
+// accessBoundary := []downscope.AccessBoundaryRule{
```

Comment:
This seems out of place, let's remove it.

Instead, this method should take in a bucket name and object name and build the access boundary. We need to show how to do this. Please refer to the access boundary rule I used in my Java PR. It will also make it easier to test.

Reply:
Or rather, cover auth_downscoping rules here